staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -111,12 +111,19 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 		// only a shorter grace period may be provided by a user
 		if options.GracePeriodSeconds != nil {
```

Comment:
this definitely needs associated unit tests exercising all the combinations of:

1. objectmeta grace period: nil, -1, 0, 1, 2
2. delete options grace period: nil, -1, 0, 1, 2